src/scanspec/core.py

```diff
@@ -20,6 +25,18 @@
 from pydantic_core import CoreSchema
 from pydantic_core.core_schema import tagged_union_schema
 
+try:
+    from types import get_original_bases
+except ImportError:
+    # function added to stdlib in 3.12
+    def get_original_bases(cls: type, /) -> tuple[Any, ...]:
+        try:
+            return cls.__dict__.get("__orig_bases__", cls.__bases__)
+        except AttributeError:
+            raise TypeError(
+                f"Expected an instance of type, not {type(cls).__name__!r}"
+            ) from None
+
```

Comment:
Can we use an explicit version check? It's easier to see when to remove when ScanSpec no longer supports <3.12

```python
if sys.version_info < (3, 12):
     ...
else:
     ...
```

Reply:
I think (once you find one) having a search term e.g. `sys.version_info` to find the rest can save time.

Reply:
Done, slight downside now is that the function isn't type-checked if it's run on a 3.12+ system